exercicios/para-casa/entregas/andrea-vetter/BankAccount.js

```diff
@@ -0,0 +1,160 @@
+import { Bank, bank1, bank2 } from "./Bank.js"
+import { Client, client1 } from "./Client.js"
+
+class BankAccount {
+  client;
+  bank;
+  accountNumber;
+  agencyNumber;
+  #balance = 0;
+
+  constructor(client, bank, accountNumber, agencyNumber) {
+    if (!(client instanceof Client)) {
+      throw "A pessoa cliente deve estar registrada no cadastro de clientes!";
+    }
+
+    if (!(bank instanceof Bank)) {
+      throw "O banco deve estar registrado no cadastro de bancos!";
+    }
+
+    this.client = client;
+    this.bank = bank;
+    this.accountNumber = accountNumber;
+    this.agencyNumber = agencyNumber;
+  }
+
+  get balance() {
+    return this.#balance;
+  }
+
+  set balance(newBalance) {
+    this.#balance = newBalance;
+  }
+
+  creditAmount(amount) {
+    this.#balance += amount;
+  }
+
+  debitAmount(amount) {
+    if (this.#balance < amount) {
+      throw new Error("Não há crédito suficiente em sua conta!");
+    }
+    this.#balance -= amount;
+  }
+
+  transferTo(anotherAccount, amount) {
+    if (!(anotherAccount instanceof BankAccount)) {
+      throw new Error("Essa conta não existe!");
+    }
+    if (this.bank === anotherAccount.bank) {
+      if (this.#balance < amount) {
+        throw `Saldo insuficiente! Você precisa de ${amount} para realizar essa transação!`;
+      }
+      this.#balance -= amount;
+      anotherAccount.#balance += amount;
+      console.log(
+        `Seu total agora é ${this.#balance} e você transferiu ${amount}.`
+      );
+    } else {
+      const bankName = this.bank;
+      const totalDebited = amount + amount * bankName.tranferTax;
+      if (this.#balance < totalDebited) {
+        throw `Saldo insuficiente! Você precisa de ${totalDebited} para realizar essa transação!`;
+      }
+      this.#balance -= totalDebited;
+      anotherAccount.#balance += amount;
+      console.log(
+        `Seu total agora é ${this.#balance} e você transferiu ${amount}.`
+      );
+    }
+  }
+
+  closeAccount() {
+    if (!(this.client instanceof Client)) {
+      throw new Error("Essa conta não existe!");
+    }
+    this.client = undefined;
+    this.bank = undefined;
+    this.accountNumber = undefined;
+    this.agencyNumber = undefined;
+  }
+
+  cashWithdrawal(amount) {
+    if (this.#balance < amount) {
```

Comment:
Sugestão: atualmente o seu debitAccount faz tudo isso, você poderia ter chamado ele, nesse caso você pode fazer dessa forma:

`this.debitAccount(amount)`

e após isso colocar o console.log de quanto saiu da conta.